README.md
`puffer-fish` let's you <b>hack this to be whatever you want</b>

![screenshot](img/languageBar.png)

<br>
<br>
<b>Look at the language data of this repo and check out the demo:</b>

<p align="center">
  <a><img src="demo1.gif" title="Demo"/></a>
</p>

<br>

<p align="center">
  <a href="https://travis-ci.org/dawsonbotsford/puffer-fish"><img src="https://travis-ci.org/dawsonbotsford/puffer-fish.svg?branch=master" alt="build status" height="22"></a>
  <a href="https://badge.fury.io/js/puffer-fish"><img src="https://badge.fury.io/js/puffer-fish.svg" alt="npm version" height="22"></a>
  <br>
  <b>Trend your Github repo on whatever language you want by generating the filetype of your choosing.</b>
</p>

<br>

## Installation
`npm install -g puffer-fish`

<br>

## Usage
```shell
puffer-fish
```

<br>

## FAQ

<br>

#### How does this help with trending?
Getting Github trending comes first from the programming language your repo is classified as. Once the [Linguist](https://github.com/github/linguist) classifies your repo into it's most prominent language, you are competing against the other popular repo's <b>also classified in that language</b> for trending.

With this hack, you can puff your repo with fake language data in order to compete for an easier language.

<br>

#### Where did the generated file go?
The generated file is placed in `.puffer-fish` which is a hidden directory. To show hidden directories, you'll need `ls -a`, `ls` does not show hidden files or directories.

This design choice was made so that puffer-fished files would remain largely hidden and out of your way during the software development process.

<br>

#### Why is the generated filename so hideous?
We used random strings in order to generate the filename in order to avoid name collisions should you want multiple puffer-fished files. With the current file length of 10 characters, this creates a possibility of 62<sup>10</sup> unique filenames. 

<br>

## Changelog
[Changelog.md](CHANGELOG.md)

<br>

## License
MIT


CHANGELOG.md
## Change Log

### upcoming (2016/05/30 06:42 +00:00)
- [#10](https://github.com/dawsonbotsford/puffer-fish/pull/10) Merge pull request #10 from dawsonbotsford/greenkeeper-mocha-2.5.0 (@dawsonbotsford)
- [e0eafa7](https://github.com/dawsonbotsford/puffer-fish/commit/e0eafa7988ca8d081c0c97eb7ed26276baa81a38) chore(package): update mocha to version 2.5.0 (@greenkeeperio-bot)
- [#11](https://github.com/dawsonbotsford/puffer-fish/pull/11) Merge pull request #11 from dawsonbotsford/greenkeeper-inquirer-1.0.3 (@dawsonbotsford)
- [0a75cf5](https://github.com/dawsonbotsford/puffer-fish/commit/0a75cf5d95e55806482e33d987692d754d6953b7) chore(package): update inquirer to version 1.0.3 (@greenkeeperio-bot)
- [103bad2](https://github.com/dawsonbotsford/puffer-fish/commit/103bad2ca560ecb97544b9bc3fa5992d6ba052b6) whitelist files and include new version in tests (@dawsonbotsford)
- [176446b](https://github.com/dawsonbotsford/puffer-fish/commit/176446b8650ffa00eb8ed9bf3052c3462e593c36) Merge branch 'master' of github.com:dawsonbotsford/puffer-fish (@dawsonbotsford)
- [ff263a5](https://github.com/dawsonbotsford/puffer-fish/commit/ff263a570dc4ad3e0b39790a2ff860f1c6ecc4d0) lighten size of module (@dawsonbotsford)
- [#5](https://github.com/dawsonbotsford/puffer-fish/pull/5) Merge pull request #5 from dawsonbotsford/greenkeeper-inquirer-0.12.0 (@dawsonbotsford)
- [3437bf7](https://github.com/dawsonbotsford/puffer-fish/commit/3437bf77cb61dfdf4d50e32782f8c359174586b1) chore(package): update inquirer to version 0.12.0 (@greenkeeperio-bot)
- [6fd60ba](https://github.com/dawsonbotsford/puffer-fish/commit/6fd60ba8d9966ee8886334a9b5cc62faafb988ab) fix url and add whitespace for npm markdown (@dawsonbotsford)
- [0316d01](https://github.com/dawsonbotsford/puffer-fish/commit/0316d015e6e5a0d6ba84e7961740b9a1ab37d0a7) cleanup readme (@dawsonbotsford)
- [c42d6fe](https://github.com/dawsonbotsford/puffer-fish/commit/c42d6fec8c449e88c19c92d1d5ee0016004b9415) Fix grammar on readme (@dawsonbotsford)
- [8b35d96](https://github.com/dawsonbotsford/puffer-fish/commit/8b35d96ec64b411bad1d2f3741be0d39c6163567) puff (@dawsonbotsford)
- [15a6907](https://github.com/dawsonbotsford/puffer-fish/commit/15a69072ee803db6621d549ff3d333a8b3e8f10e) 1.0.0 release (@dawsonbotsford)

### v1.0.0 (2016/02/02 19:18 +00:00)
- [4171dba](https://github.com/dawsonbotsford/puffer-fish/commit/4171dbad5e94ee9c4a292f5f84fec861fb8983c6) improve faq, etc. (@dawsonbotsford)
- [0b4928e](https://github.com/dawsonbotsford/puffer-fish/commit/0b4928ef22acbbcc9ef1d9d16eec6e88e74aa7aa) Generate random filename. Closes #1 (@dawsonbotsford)
- [e7a8114](https://github.com/dawsonbotsford/puffer-fish/commit/e7a811469fd72aa8c5fedabe443de887f364918b) fix indentation (@dawsonbotsford)
- [f7e422e](https://github.com/dawsonbotsford/puffer-fish/commit/f7e422ecde84cef46864e21cba1acddff8d1f056) make hidden puffer-dir if it does not already exist (@dawsonbotsford)
- [943d1f6](https://github.com/dawsonbotsford/puffer-fish/commit/943d1f60d9e29e7f9b61ad156341dd51ba27b29c) functioning beta! (@dawsonbotsford)
- [bb0bc81](https://github.com/dawsonbotsford/puffer-fish/commit/bb0bc81caf3df1fe92450e3f9c418cc758e276c6) Better output (@dawsonbotsford)
- [e750681](https://github.com/dawsonbotsford/puffer-fish/commit/e750681876e9c1c1e9b01ea2ce85b856508e5b96) Saves to file (@dawsonbotsford)
- [75c568a](https://github.com/dawsonbotsford/puffer-fish/commit/75c568a5b97772305f1ff78f2e9805048d6f7a59) cli questioning init (@dawsonbotsford)
- [8dcefae](https://github.com/dawsonbotsford/puffer-fish/commit/8dcefae75790178414f398523c408b8db3c971ad) strong readme init (@dawsonbotsford)
- [3907678](https://github.com/dawsonbotsford/puffer-fish/commit/39076780726256828afb44dd3c9161cc8d2fc812) package init (@dawsonbotsford)
- [c85bc3e](https://github.com/dawsonbotsford/puffer-fish/commit/c85bc3e020aaf88e876f438b8b5c2697d784e2e6) Initial commit (@dawsonbotsford)

test/helper.spec.js
const should = require('chai').should();
const expect = require('chai').expect;
const	helper = require('../helper');

const template = [{'JavaScript': 'js'}];

describe('#buildChoices', function() {
  it('returns an array', function() {
		const result = helper.buildChoices(template);
		expect(result).be.a('array');
  });
});

describe('#getSuffix', function() {
  it('returns correct suffix from most popular', function() {
		const result = helper.getSuffix('JavaScript');
		result.should.equal('js');
  });
  it('returns correct suffix from moderately popular', function() {
		const result = helper.getSuffix('CSS');
		result.should.equal('css');
  });
  it('returns correct suffix from most popular', function() {
		const result = helper.getSuffix('assembly');
		result.should.equal('nasm');
  });
});

describe('#isSafeInteger', function() {
  it('returns a boolean', function() {
    const result = helper.isSafeInteger(1);
    expect(result).be.a('boolean');
  });
  it('float returns false', function() {
    const result = helper.isSafeInteger('.1');
    result.should.equal(false);
  });
  it('small int returns true', function() {
    const result = helper.isSafeInteger('9');
    result.should.equal(true);
  });
  it('large int returns false', function() {
    const result = helper.isSafeInteger('999999999');
    result.should.equal(false);
  });
  it('negative int returns false', function() {
    const result = helper.isSafeInteger('-9');
    result.should.equal(false);
  });
  it('0 returns false', function() {
    const result = helper.isSafeInteger('0');
    result.should.equal(false);
  });
});


helper.js
'use strict';

const languages = require('./languages.json');
const inquirer = require('inquirer');
const fs = require('fs');

module.exports = {
  getSuffix: function(lang) {
    let toReturn = '';
    languages.forEach(function (elem, index) {
      if (typeof elem === 'object') {
        if (languages[index].hasOwnProperty(lang)){
          toReturn = languages[index][lang];
        }
      }
      //else throw error
    });
    return toReturn;
  },

  buildChoices: function() {
    let myChoices = [];
    for (let elem of languages) {
      if (typeof elem === 'string') {
        myChoices.push(new inquirer.Separator(' '));
        myChoices.push(new inquirer.Separator(elem));
      }
      else {
        for (let i in elem){
          myChoices.push(i);
        }
      }
    }
    return myChoices;
  },

  /*
   * arg1 must be > 0 and < 300,000,000
   * */
  isSafeInteger: function(str) {
    var n = ~~Number(str);
    return String(n) === str && n > 0 && n <= 300000000;
  },

  fileExists: function(fullPath) {
    let toReturn = false;
    fs.accessSync(fullPath, fs.R_OK | fs.W_OK, function(err) {
      if (!err) {
        toReturn = true;
      }
    });
    return toReturn;
  }

};


puffer-fish.sh
pppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppp

cli.js
#!/usr/bin/env node
'use strict';

const inquirer = require('inquirer');
const fs = require('fs');
const chalk = require('chalk');
const helper = require('./helper.js');
const rs = require('random-strings');
const pb = require('pretty-bytes');

inquirer.prompt([
  {
    type: 'list',
    name: 'language',
    message: 'What language would you like to generate?',
    default: 'CSS',
    choices: helper.buildChoices()
  },
  {
    type: 'input',
    name: 'size',
    message: 'What size file? (bytes)',
    default: '10000',
    filter: function( val ) {
      if (!helper.isSafeInteger(val)) {
        throw new Error('File size must be a positive integer <= 300,000,000');
      }
      return val.toLowerCase();
    }
  }
], function( answers ) {
  inquirer.prompt([
    {
      type: 'confirm',
      name: 'generateFile',
      message: 'Generate a ' + pb(Number(answers.size)) + ' ' + answers.language + ' file?',
      default: 'true'
    }
  ], function( confirm ) {
    if (confirm.generateFile === true) {
      const suffix = helper.getSuffix(answers.language);
      const pufferFishDir = '.puffer-fish';
      const fileName = rs.alphaNumMixed(10) + '.' + suffix;
      const fullPath = pufferFishDir + '/' + fileName;

			//build pufferFishDir for the puffer files if it does not already exist
			try {
					const stats = fs.lstatSync(pufferFishDir);

					if (stats.isDirectory()) {
						chalk.green(console.log(pufferFishDir + ' dir already exists! Thanks for coming back.'));
					}
					else {
						console.log('Creating ' + pufferFishDir + ' for your files...');
						fs.mkdirSync(pufferFishDir);
					}
			}
			catch (e) {
				console.log('Creating ' + pufferFishDir + ' for your files...');
				fs.mkdirSync(pufferFishDir);
			}

      const finalString = new Array( Number(answers.size) + 1).join('p');

      fs.writeFileSync(fullPath, finalString, 'utf8');

      console.log(chalk.green('File successfully written to ' + fullPath));
    }

    else {
      console.log(chalk.yellow('No file puffer-fished'));
    }

  });
});
